samcli/local/apigw/local_apigw_service.py

```diff
@@ -604,29 +636,42 @@ def _request_handler(self, **kwargs):
             headers = Headers(cors_headers)
             return self.service_response("", headers, 200)
 
-        if isinstance(route.authorizer_object, LambdaAuthorizer) and not self._valid_identity_sources(route):
+        # check for LambdaAuthorizer since that is the only authorizer we currently support
+        if isinstance(lambda_authorizer, LambdaAuthorizer) and not self._valid_identity_sources(route):
             return ServiceErrorResponses.missing_lambda_auth_identity_sources()
 
         try:
-            event = self._generate_lambda_event(request, route, method, endpoint)
+            route_lambda_event = self._generate_lambda_event(request, route, method, endpoint)
+            auth_lambda_event = None
+
+            if lambda_authorizer:
+                auth_lambda_event = self._generate_lambda_authorizer_event(request, route, lambda_authorizer)
         except UnicodeDecodeError as error:
             LOG.error("UnicodeDecodeError while processing HTTP request: %s", error)
             return ServiceErrorResponses.lambda_failure_response()
 
-        stdout_stream = io.BytesIO()
-        stdout_stream_writer = StreamWriter(stdout_stream, auto_flush=True)
-
         try:
-            self.lambda_runner.invoke(route.function_name, event, stdout=stdout_stream_writer, stderr=self.stderr)
+            if lambda_authorizer:
```

Comment:
Just my opinion, but I think we can move some of this specific logic to a separate function:
```
if lambda_authorizer:
      do_authorizer_logic(context)
... 
```
just so these functions don't keep getting so bulky.